traffic_ops/install/bin/postinstall.py

```diff
@@ -0,0 +1,1344 @@
+#!/usr/bin/env -S sh -c 'PATH="${PATH}:/usr/libexec" exec "$(type -p python platform-python | head -n1)" "$0" "$@"'
```

Comment:
Testing it out, ^ needs a semicolon after the PATH addition in order to work, like

```shell
#!/usr/bin/env -S sh -c 'PATH="${PATH}:/usr/libexec"; exec "$(type -p python3 platform-python | head -n1)" "$0" "$@"'
```

However, apparently `env -S` is not supported on the GNU `env` that CentOS 7 uses (8.22), and since CentOS 8 goes EOL 2.5 years before CentOS 7 does, we should support CentOS 7's `env`. This would work:

```bash
#!/usr/bin/env bash
"exec" "bash" "-c" "PATH+=:/usr/libexec/ exec \$(type -p python3 platform-python | head -n1) \"$0\" $*"
```

Reply:
Turning a Python script into a polyglot bash script seems convoluted. Can I not just use `#!/usr/bin/env python3` or `#!/bin/sh -c 'PATH="${PATH}:/usr/libexec"; exec "$(type -p python3 platform-python | head -n1)" "$0" "$@"'` ? Why does `env` need so much hand-holding? If `platform-python` isn't in the `PATH` should I be using it at all? [The RedHat Developer Blog](https://developers.redhat.com/blog/2018/11/14/python-in-rhel-8/) says no:

> _"If you are not re-building the distro, do not use Platform-Python directly. Install `python3` and use that."_

and later reiterates:

> _"Do not use platform-python for your applications. However, use platform-python if you are writing system/admin code for RHEL 8."_

Reply:
> Turning a Python script into a polyglot bash script seems convoluted. Can I not just use `#!/usr/bin/env python3` or `#!/bin/sh -c 'PATH="${PATH}:/usr/libexec"; exec "$(type -p python3 platform-python | head -n1)" "$0" "$@"'` ? Why does `env` need so much hand-holding?

It's not `env` that needs the handholding, it's shebang. Shebang does not split arguments by a *space* character (or by anything else), so in the snippet you posted above, `-c 'PATH="${PATH}:/usr/libexec"; exec "$(type -p python3 platform-python | head -n1)" "$0" "$@"'` is all treated as a single argument, which results in an error:

```shell
[root@732dff90b85f a]# ./a.py
/bin/sh: - : invalid option
Usage:  /bin/sh [GNU long option] [option] ...
        /bin/sh [GNU long option] [option] script-file ...
GNU long options:
        --debug
        --debugger
        --dump-po-strings
        --dump-strings
        --help
        --init-file
        --login
        --noediting
        --noprofile
        --norc
        --posix
        --rcfile
        --rpm-requires
        --restricted
        --verbose
        --version
Shell options:
        -ilrsD or -c command or -O shopt_option         (invocation only)
        -abefhkmnptuvxBCHP or -o option
```


> If `platform-python` isn't in the `PATH` should I be using it at all? [The RedHat Developer Blog](https://developers.redhat.com/blog/2018/11/14/python-in-rhel-8/) says no:
> 
> > _"If you are not re-building the distro, do not use Platform-Python directly. Install `python3` and use that."_
> 
> and later reiterates:
> 
> > _"Do not use platform-python for your applications. However, use platform-python if you are writing system/admin code for RHEL 8."_

Since there is no Python 2 support currently, yeah, might as well just use `#!/usr/bin/env python3` directly for now.

Reply:
> _"Since there is no Python 2 support currently, yeah, might as well just use `#!/usr/bin/env python3` directly for now."_

Then that's what I'll do